lib/mix/tasks/ash_authentication.install.ex

```diff
@@ -0,0 +1,308 @@
+      {:ok, {igniter, _, _}} ->
+        {:ok,
+         Igniter.add_warning(
+           igniter,
+           "Token resource already exists: #{token_resource}, skipping creation."
+         )}
+
+      {:error, igniter} ->
+        extensions = "AshAuthentication.TokenResource,Ash.Policy.Authorizer"
+
+        extensions =
+          if Code.ensure_loaded?(AshPostgres.DataLayer) do
+            "postgres,#{extensions}"
+          else
+            extensions
+          end
+
+        igniter
+        |> Igniter.compose_task(
+          "ash.gen.resource",
+          [
+            inspect(token_resource),
+            "--extend",
+            extensions,
+            "--attribute",
+            "jti:string:primary_key:public:required:sensitive",
+            "--attribute",
+            "subject:string:required",
+            "--attribute",
+            "expires_at:utc_datetime:required",
+            "--attribute",
+            "purpose:string:required:public",
+            "--attribute",
+            "extra_data:map:public",
+            "--timestamps"
+          ] ++ argv ++ resource_args
+        )
+        # Consider moving to the extension's `install/5` callback, but we need
+        # to only run it if the resource is being created which we can't
+        # currently tell in that callback
+        |> Ash.Resource.Igniter.add_action(token_resource, """
+        read :expired do
+          filter expr(expires_at < now())
+        end
+        """)
+        |> Ash.Resource.Igniter.add_bypass(
+          token_resource,
+          quote do
+            AshAuthentication.Checks.AshAuthenticationInteraction
+          end,
+          quote do
+            authorize_if always()
+          end
+        )
+        |> Ash.Resource.Igniter.add_policy(
+          token_resource,
+          quote do
+            always()
+          end,
+          quote do
+            forbid_if always()
+          end
+        )
+        |> Ash.Resource.Igniter.add_action(token_resource, """
+        read :get_token do
+          get? true
+          argument :token, :string, sensitive?: true
+          argument :jti, :string, sensitive?: true
+          argument :purpose, :string, sensitive?: false
+
+          prepare AshAuthentication.TokenResource.GetTokenPreparation
+        end
+        """)
+        |> Ash.Resource.Igniter.add_action(token_resource, """
+        read :revoked? do
+          argument :token, :string, sensitive?: true
+          argument :jti, :string, sensitive?: true
+          get? true
+
+          prepare AshAuthentication.TokenResource.IsRevokedPreparation
+        end
+        """)
+        |> Ash.Resource.Igniter.add_action(token_resource, """
+        read :get_confirmation_changes do
+          argument :jti, :string, allow_nil?: false, sensitive?: true
+          get? true
+
+          prepare AshAuthentication.TokenResource.GetConfirmationChangesPreparation
+        end
+        """)
```

Comment:
these (`*_confirmation_*`) actions are only required if the confirmation add-on has been added.

Reply:
Interesting...I don't see them being generated conditionally under the hood. Am I missing something there?

Reply:
If they're not that's an oversight. Also doesn't help that they've been essentially invisible until now.

Reply:
I think actually it's because you can't look at other resource information in transformers. So you can't know if you need to add them or not.